napari/layers/utils/_link_layers.py

```diff
@@ -223,7 +224,17 @@ def _get_common_evented_attributes(
             )
         )
 
-    common_events = set.intersection(*(set(lay.events) for lay in layers))
+    layer_events = []
+    for lay in layers:
+        # skip deprecated events
+        layer_events.append(
+            {
+                ev
+                for ev in lay.events
+                if not isinstance(lay.events[ev], WarningEmitter)
+            }
+        )
```

Comment:
```suggestion
    layer_events = [
        {
            e for e in lay.events
            if not isinstance(lay.events[e], WarningEmitter)
        }
        for lay in layers
    ]
```

Optional: I marginally prefer the list comprehension, even with the inner set comprehension. But I'll leave this up to you.